test_framework/basic.py

```diff
@@ -399,8 +405,11 @@ def compile_and_run(self, source_file: Path) -> None:
         print_stderr(compile_result)
 
         # run the executable
-        # TODO cleaner handling if executable doesn't exist? or check that it exists above?
```

Comment:
As I stumbled over it I fixed this on the way. Using an assertion makes it go through the stacktrace-filtering approach instead of raising the FileNotFoundError :)